dubbo-common/src/main/java/org/apache/dubbo/common/logger/LoggerFactory.java

```diff
@@ -34,40 +36,43 @@
  */
 public class LoggerFactory {
 
-    private static final ConcurrentMap<String, FailsafeLogger> LOGGERS = new ConcurrentHashMap<String, FailsafeLogger>();
+    private static final ConcurrentMap<String, FailsafeLogger> LOGGERS = new ConcurrentHashMap<>();
     private static volatile LoggerAdapter LOGGER_ADAPTER;
 
     // search common-used logging frameworks
     static {
-        String logger = System.getProperty("dubbo.application.logger");
-        if ("slf4j".equals(logger)) {
-            setLoggerAdapter(new Slf4jLoggerAdapter());
-        } else if ("jcl".equals(logger)) {
-            setLoggerAdapter(new JclLoggerAdapter());
-        } else if ("log4j".equals(logger)) {
-            setLoggerAdapter(new Log4jLoggerAdapter());
-        } else if ("jdk".equals(logger)) {
-            setLoggerAdapter(new JdkLoggerAdapter());
-        } else if ("log4j2".equals(logger)) {
-            setLoggerAdapter(new Log4j2LoggerAdapter());
-        } else {
-            try {
+        String logger = System.getProperty("dubbo.application.logger", "");
+        switch (logger) {
+            case "slf4j":
+                setLoggerAdapter(new Slf4jLoggerAdapter());
+                break;
+            case "jcl":
+                setLoggerAdapter(new JclLoggerAdapter());
+                break;
+            case "log4j":
                 setLoggerAdapter(new Log4jLoggerAdapter());
-            } catch (Throwable e1) {
-                try {
-                    setLoggerAdapter(new Slf4jLoggerAdapter());
-                } catch (Throwable e2) {
+                break;
+            case "jdk":
+                setLoggerAdapter(new JdkLoggerAdapter());
+                break;
+            case "log4j2":
+                setLoggerAdapter(new Log4j2LoggerAdapter());
+                break;
+            default:
+                List<Class<? extends LoggerAdapter>> candidates = Arrays.asList(
+                        Log4jLoggerAdapter.class,
+                        Slf4jLoggerAdapter.class,
+                        Log4j2LoggerAdapter.class,
```

Comment:
Is it need 3 other log class:  Log4j2LoggerAdapter, JclLoggerAdapter, JdkLoggerAdapter.

Reply:
![image](https://user-images.githubusercontent.com/15965696/51969315-0c8db100-24af-11e9-9bf7-eedaa1f69729.png)
What do you mean, did I miss anything out?